test/reactive-handler.spec.ts

```diff
@@ -679,4 +663,139 @@ describe('ReactiveHandler', () => {
+            expect(target.getProxy(value)).toBe(value);
+        })
+    });
+
+    describe('handles array type values', () => {
+        let defaultMembrane;
+        beforeAll(() => {
+            defaultMembrane = new ReactiveMembrane();
+        });
+        it('wraps array', () => {
+            const value = []
+            const proxy = defaultMembrane.getProxy(value);
+            expect(proxy).not.toBe(value);
+            expect(defaultMembrane.unwrapProxy(value)).toBe(value);
+        });
+        it('access items in array', () => {
+            const value = ['foo', 'bar'];
+            const proxy = defaultMembrane.getProxy(value);
+            expect(Array.isArray(proxy)).toBe(true);
+            expect(proxy.length).toBe(2);
+            expect(proxy[0]).toBe('foo');
+            expect(proxy[1]).toBe('bar');
+        });
+        it('should notify when values accessed', () => {
+            const value = ['foo', {nested: 'bar'}];
+            const accessSpy = jest.fn();
+            const observedMembrane = new ReactiveMembrane({
+                valueObserved: accessSpy,
+            });
+    
+            const proxy = observedMembrane.getProxy(value);
+            doNothing(proxy[0]);
+            doNothing(proxy[1]);
+            expect(accessSpy).toHaveBeenCalledTimes(2);
+            expect(accessSpy).toHaveBeenCalledWith(value, '0');
+            expect(accessSpy).toHaveBeenCalledWith(value, '1');
+        });
+
+        describe('access values in array', () => {
+            let membrane;
+            const accessSpy = jest.fn();
+            const changeSpy = jest.fn();
+            beforeAll(() => {
+                membrane = new ReactiveMembrane({
+                    valueObserved: accessSpy,
+                    valueMutated: changeSpy
+                });
+            });
+            afterEach(() => {
+                accessSpy.mockClear();
+                changeSpy.mockClear();
+            });
+            it('entries() works on wrapped array', () => {
+                const value = ['foo', 'bar'];
+                const proxy = membrane.getProxy(value);
+                const entries = proxy.entries();
+                expect(entries.next().value).toEqual([0, 'foo']);
+                expect(entries.next().value).toEqual([1, 'bar']);
+                expect(entries.next().done).toBe(true);
+                // 6 invocations: 1 time for .entries, 3 times for entries.next() and 2 times for actually accessing the index value
+                expect(accessSpy).toHaveBeenCalledTimes(6);
+                expect(changeSpy).toHaveBeenCalledTimes(0);
+            });
+            it('concat() does not notify a mutation', () => {
+                const value = ['foo', 'bar'];
+                const proxy = membrane.getProxy(value);
+                const actual = proxy.concat(['baz']);
+                //  May be there is some optimization we can do here? Does valueObserved need to be called so many times?
```

Comment:
Here is an example of a simplified membrane: https://jsfiddle.net/pmdartus/preb6vtq/6/
It is expected to call access 8 times if you follow along with the spec side by side.

Reply:
@pmdartus Totally understand why there are 8 calls to the proxy. My question was whether the membrane can be intelligent about when to call valueObserved(). 
In the current model, the handler is a simple publisher. Its up to the subscriber(observer) to decide to act or not. 
This model is probably fine. Just wanted to bring it up and see if it is a concern at all.